### ISSUE TITLE
Possible to configure $ vs unprefixed?

### ISSUE THREAD

**Issue by kirbysayshi** [altano/handlebars-loader#42]
Hello,

I'm investigating using this loader, but in our current templates we use the following semantics for partials:

```
{{> some/tpl.hbs}} -> look in node_modules/some/tpl.hbs (aka use node_modules resolution algo)
```
```
{{> ./some/tpl.hbs}} -> look in ./some/tpl.hbs
```

Is it possible (meaning as designed) to configure this loader to do the same, which is basically to behave with the same semantics of node's node_modules algorithm?

Thanks!

**Comment by altano** [comment]
When you prefix the partial (or whatever) with '$', handlebars-loader will use webpack's loader API for resolving modules, which will include looking in node_modules.  But judging from the title of the question, I figure you already knew that, and I don't really understand the question.  Are you asking if it is possible to have the behavior of '$' but without the prefix?

**Comment by kirbysayshi** [comment]
Yes! Where by default any partial will be resolved using the loader api.

Yip yip,
~ Drew*

**Comment by altano** [comment]
Actually I just took a quick look at the code and it looks like the documentation might be wrong/out-of-date.  sokra originally wrote the majority of the plug-in and I think the '$' prefix was used then, but it looks like in the current coad the loader API .resolve() method is always used.  Have you given it a shot and it doesn't work?  Or are you just going by the documentation?

If you have code that isn't working, aka is working as the documentation states, can you link me to some code so I can take a look?

**Comment by kirbysayshi** [comment]
Honestly, I looked at the documentation, then looked at the code a bit. I
didn't actually try myself, as I was investigating if webpack would work at
a very high level (and had some other problems to solve as well for this
project).

If I have some time, I'll try to write some code to test out the current
behavior.

Thanks so much for responding!

Yip yip,
~ Drew*

**Comment by altano** [comment]
I just gave it a quick test and it doesn't work actually, so I'll look at this issue when I have some free time to reacquaint myself with the code.

**Comment by diurnalist** [comment]
The resolve behavior is weird in my opinion. Most of the time it tries to look up relative to the template file itself. I would also prefer that it only did that if you explicitly prefixed your thing w/ a relative path, like your example:

```handlebars
{{> ./some/tpl.hbs}}
```

If we changed this behavior (which I would support!) it would obviously be a breaking change. It would be nice to expose some option in the loader to allow people to retain the old behavior, though. Not sure what that looks like.

So you know, it will follow the webpack resolve algorithm (looking in modules folders etc) if:

- the thing (helper, partial, etc) is prefixed with `$`
- the thing is a helper, and you've defined helperDirs

**Comment by kirbysayshi** [comment]
Does it even need a $ to trigger node_modules resolution? I've seen other loaders work fine with the following rules:

```
{{> ./this/is/a/relative/path.hbs}}
{{> this/is/a/node_module.hbs}}
{{> $this/is/something/special.hbs}}
```

And then wouldn't a helper not use the `{{>` prefix, and instead use `{{`? My handlebars knowledge isn't the best so I could be wrong.

**Comment by kirbysayshi** [comment]
The rules above are basically the exact same that node itself uses, with the addition of the `$`.
